File: examples/example4.py
#!/usr/bin/env python
import sys
sys.path.insert(0,'..')
import drqueue
import os

print "Master: %s\n"%(os.environ["DRQUEUE_MASTER"])

# First we receive the list of slaves
computer_list = drqueue.request_computer_list (drqueue.CLIENT)
print "Computers connected to the master:"
for computer in computer_list:
    # We can check if the computer is enabled looking into computer.limits.enabled (0 if disabled)
    print "Computer: %s Enabled: %s"%(computer.hwinfo.name,(lambda x: x and "Yes" or "No")(computer.limits.enabled))
    if computer.limits.enabled:
        # If the computer is enabled we disable it
        computer.request_disable(drqueue.CLIENT)
        print "\tDisabled !"
    else:
        # And if it is disabled we enable it
        computer.request_enable(drqueue.CLIENT)
        print "\tEnabled !"

File: ruby/example4.rb
#!/usr/bin/ruby


require 'drqueue'


puts "Master: #{ENV["DRQUEUE_MASTER"]}\n"

# First we receive the list of slaves
computer_list = Drqueue::request_computer_list(Drqueue::CLIENT)
puts "Computers connected to the master:"
computer_list.each do |c|
	# We can check if the computer is enabled looking into computer.limits.enabled (0 if disabled)
	print "Computer: #{c.hwinfo.name} | Enabled: "
	if "#{c.limits.enabled}" == "1"
		print "Yes"
		# If the computer is enabled we disable it
		c.request_disable(Drqueue::CLIENT)
		print "\tDisabled !\n"
	else
		print "No"
		# If the computer is disabled we enable it
		c.request_enable(Drqueue::CLIENT)
		print "\tEnabled !\n"
	end
end

